### ISSUE TITLE
LBRY Debian / Apt / Ubuntu Install Cleanup

### ISSUE THREAD

**Issue by kauffj** [lbryio/lbry-app#1376]
There are several issues with the LBRY experience in the Ubuntu Software Center:

[ ] An extraneously version appears from the "SnapStore", v0.14
[ ] No screenshot appears for the app
[ ] The license shows as proprietary, when it is actually MIT

I'm guessing the bottom two are fixed by setting correct metadata... somewhere.

**Comment by kauffj** [comment]
This is still true. In particular I'd like at least the extremely old, broken version to not show up when I search "LBRY" (it's the first result).

I'm guessing the other 2 are just properly setting some metadata in one of the build artifacts.

**Comment by kauffj** [comment]
Actually, the "correct" entry is also wrong. It's showing 0.21.5, not 0.23.

**Comment by tzarebczan** [comment]
Re license:
https://github.com/electron-userland/electron-builder/issues/3914

Could be an existing bug: https://bugs.launchpad.net/ubuntu/+source/software-center/+bug/435183 / https://github.com/betaflight/betaflight-configurator/issues/989#issuecomment-375954366

https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/

**Comment by tiger5226** [comment]
The proprietary license issue was brought up yesterday by a friend who was trying out lbry. Probably very low priority but wanted to flag that it was noticed.

**Comment by kauffj** [comment]
https://github.com/electron-userland/electron-builder/issues/4270

**Comment by kauffj** [comment]
I've found like a dozen people/projects that have this issue and no one seems to know how to get it to say anything other than proprietary.

**Comment by tiger5226** [comment]
it looks like we were building for uappexplorer which is shutdown now. I do not see lbry on the ubuntu snap store. This is something different to build to. 

looking here https://github.com/lbryio/lbry-desktop/blob/master/electron-builder.json#L66

We only specify `deb` for the top level `linux` key. See docs below, but I think we should add `snap` as well. 
https://www.electron.build/configuration/linux

And the details for `snap` are

**Comment by kauffj** [comment]
@tiger5226 yeah, it may be better to just pull the trigger and switch to snaps.

**Comment by kauffj** [comment]
or appimages or flatpak or...

open-source is both fun and terrible :grin:

**Comment by danrobi11** [comment]
Here: https://bugs.launchpad.net/ubuntu/+source/software-center/+bug/435183

**Comment by danrobi11** [comment]
I got something from Lubuntu: 
"I couldn’t found those packages in ubuntu repositories, so maybe there are some issues with the way the .deb was created. I downloaded FreeTube to look at it ( https://github.com/FreeTubeApp/FreeTube/releases/download/v0.7.1-beta/FreeTube_0.7.1_amd64.deb ) and couldn’t found a /debian folder in which, among others, the license in a machine readable format (copyright) should be placed. Maybe that’s why the system assume it is propietary. More info of copyright file: https://www.debian.org/doc/packaging-manuals/copyright-format/1.0/"

Source: https://discourse.lubuntu.me/t/deb-packages-license-issue/470/4

**Comment by kauffj** [comment]
Great find @danrobi11!

**Comment by danrobi11** [comment]
> Got new comment from Lubuntu:
> Source: https://discourse.lubuntu.me/t/deb-packages-license-issue/470/16
> 
> > I don’t see that any of these packages have an appstream.xml or metainfo.xml as specified by the [AppStream specification](https://www.freedesktop.org/software/appstream/docs/sect-Metadata-Application.html) and their desktop entries are not readily found (another place metadata could come from). That also may be where the problem lies.
> 
> Or it could be a combination of these things. Either way, this illustrates a good point: software in the Ubuntu archives is carefully checked to ensure it meets a variety of different specifications to ensure it works properly with the whole ecosystem of applications available in the archives. When you’re using software from outside the archives, you’re bound to run into trouble. This sort of software is entirely unsupported by Ubuntu.
> 
> If you want this to work correctly, you will need to talk to the upstream developers and perhaps have them speak with the gnome-software developers to figure out the appropriate solution. Additionally, you should urge them to get their packages included into Debian. Those packages should get automatically synced to Ubuntu.

**Comment by danrobi11** [comment]
Like wxl at Lubuntu suggested, i've opened a deb packages license issue at gnome-software gitlab
https://gitlab.gnome.org/GNOME/gnome-software/issues/833

**Comment by danrobi11** [comment]
Andre Klapper at the Gnome-software needs more information. Please have a look: https://gitlab.gnome.org/GNOME/gnome-software/issues/833#note_628721

**Comment by pktiuk** [comment]
> The license shows as proprietary, when it is actually MIT

`.deb` package for lbry is not very good, it lacks some informations.

To check what is wrong I recommend lintian.
<br>Current output:
```bash
lintian ./LBRY_0.47.1.deb 
E: lbry: debian-changelog-file-missing-or-wrong-name
E: lbry: description-synopsis-is-empty
E: lbry: dir-or-file-in-opt opt/LBRY/
E: lbry: dir-or-file-in-opt opt/LBRY/LICENSE.electron.txt
E: lbry: dir-or-file-in-opt opt/LBRY/LICENSES.chromium.html
E: lbry: dir-or-file-in-opt ... use --no-tag-display-limit to see all (or pipe to a file/program)
E: lbry: embedded-library opt/LBRY/lbry: freetype
E: lbry: embedded-library opt/LBRY/lbry: lcms2
E: lbry: embedded-library opt/LBRY/lbry: libjpeg
E: lbry: embedded-library ... use --no-tag-display-limit to see all (or pipe to a file/program)
E: lbry: missing-dependency-on-libc needed by opt/LBRY/lbry and 7 others
E: lbry: no-copyright-file
E: lbry: shlib-with-executable-bit opt/LBRY/libEGL.so 0755
E: lbry: shlib-with-executable-bit opt/LBRY/libGLESv2.so 0755
E: lbry: shlib-with-executable-bit opt/LBRY/libVkICD_mock_icd.so 0755
E: lbry: shlib-with-executable-bit ... use --no-tag-display-limit to see all (or pipe to a file/program)
E: lbry: statically-linked-binary opt/LBRY/resources/static/lbry-first/lbry-first
E: lbry: unstripped-binary-or-object opt/LBRY/libVkICD_mock_icd.so
E: lbry: unstripped-binary-or-object opt/LBRY/resources/static/lbry-first/lbry-first
E: lbry: unstripped-binary-or-object opt/LBRY/swiftshader/libEGL.so
E: lbry: unstripped-binary-or-object ... use --no-tag-display-limit to see all (or pipe to a file/program)
W: lbry: description-starts-with-leading-spaces
...
```

**Comment by Victor239** [comment]
Flatpak is better than Snap because:
- Server side code for Snap is closed
- Snap doesn't allow users to connect to more than one server. This is similar to Apple's monopoly on the iOS App Store, which means that nobody else can host a server for users to download apps from.
